pkg/apis/batch/interfaces.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package batch
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+)
+
+func (obj *Job) PodSpec() (*api.PodSpec, string) {
```

Comment:
> I vote against doing it this way. I would like to get rid of the internal types and this makes it harder.
> 
> Instead why not just centralize the big switch statement somewhere?

To do that and allow an API group to provide its own `PodSpecer` types would require plugin-style registration.  Would I get to have that?

I started this on the internal types (happens to be what we use today), but nothing precludes having this on the external type.